tests/run.sh

```diff
@@ -17,10 +17,14 @@ if [ "$CI" = true ]; then
     set -x
 fi
 
-source_dir="$PWD"
+source_dir="$(dirname "$(realpath "$(type -p $0)")")"/..
```

Comment:
I don't think `type -0` is portable. My ZSH has different output than bash (zsh says e.g. `/usr/bin/test is /usr/bin/test`). Since this shell script runs with `/bin/sh`, that's a problem.

Also, right now I don't really get the problem that this tries to fix. AFAIK my patch to this script was "if $CMAKE_BINARY_DIR is set, use that, else do exactly the same thing as before my patch"